Detectors/CTP/workflowIO/src/DigitReaderSpec.cxx

```diff
@@ -58,29 +62,79 @@ class DigitReader : public Task
 DigitReader::DigitReader(bool useMC)
 {
   if (useMC) {
-    LOG(info) << "CTP does not support MC truth at the moment";
+    LOG(info) << "CTP : truth = data as CTP inputs are already digital";
   }
 }
 
 void DigitReader::init(InitContext& ic)
 {
   auto filename = o2::utils::Str::concat_string(o2::utils::Str::rectifyDirectory(ic.options().get<std::string>("input-dir")),
                                                 ic.options().get<std::string>("ctp-digit-infile"));
+  LOG(info) << "init: " << ic.options().hasOption("ignore-irframes") << " " << ic.options().get<bool>("ignore-irframes");
+  if (ic.options().hasOption("ignore-irframes") && !ic.options().get<bool>("ignore-irframes")) {
+    mUseIRFrames = true;
+  }
   connectTree(filename);
 }
 
 void DigitReader::run(ProcessingContext& pc)
 {
-  auto ent = mTree->GetReadEntry() + 1;
-  assert(ent < mTree->GetEntries()); // this should not happen
-
-  mTree->GetEntry(ent);
-  LOG(info) << "DigitReader pushes " << mDigits.size() << " digits at entry " << ent;
-  pc.outputs().snapshot(Output{"CTP", "DIGITS", 0}, mDigits);
-  pc.outputs().snapshot(Output{"CTP", "LUMI", 0}, mLumi);
-  if (mTree->GetReadEntry() + 1 >= mTree->GetEntries()) {
-    pc.services().get<ControlService>().endOfStream();
-    pc.services().get<ControlService>().readyToQuit(QuitRequest::Me);
+  gsl::span<const o2::dataformats::IRFrame> irFrames{};
+  // LOG(info) << "Using IRs:" << mUseIRFrames;
+  if (mUseIRFrames) {
+    irFrames = pc.inputs().get<gsl::span<o2::dataformats::IRFrame>>("driverInfo");
+  }
+  auto ent = mTree->GetReadEntry();
+  if (!mUseIRFrames) {
+    ent++;
+    assert(ent < mTree->GetEntries()); // this should not happen
+    mTree->GetEntry(ent);
+    LOG(info) << "DigitReader pushes " << mDigits.size() << " digits at entry " << ent;
+    pc.outputs().snapshot(Output{"CTP", "DIGITS", 0}, mDigits);
+    pc.outputs().snapshot(Output{"CTP", "LUMI", 0}, mLumi);
+    if (mTree->GetReadEntry() + 1 >= mTree->GetEntries()) {
+      pc.services().get<ControlService>().endOfStream();
+      pc.services().get<ControlService>().readyToQuit(QuitRequest::Me);
+    }
+  } else {
+    std::vector<o2::ctp::CTPDigit> digitSel;
+    if (irFrames.size()) { // we assume the IRFrames are in the increasing order
+      if (ent < 0) {
+        ent++;
+      }
+      o2::utils::IRFrameSelector irfSel;
+      long biasInBC = 0; // depends if digits are already aligned
+      irfSel.setSelectedIRFrames(irFrames, 0, 0, -biasInBC, true);
```

Comment:
In the case of the CTP the bias is 0, you could simply pass it.

Reply:
removed